### ISSUE TITLE
Error message on "Do so by passing the top qualifier from the given hierarchy." violation is not informative

### ISSUE THREAD

**Issue by PukPukov** [typetools/checker-framework#7652]
If you create custom type hierarchy with subtyping checker, add polymorphic qualifier and pass to PolymorphicQualifier meta-annotation value() property class that is not top qualifier, you will get TypeSystemError that says something about bottom types, rather than that you have passed not the top qualifier. Debugging this can be quite tedious.

Example:

```java
@DefaultQualifierInHierarchy
@SubtypeOf({})
@Target({ElementType.TYPE_USE, ElementType.TYPE_PARAMETER})
public @interface root { }

@SubtypeOf({
    foo.class, boo.class, goo.class
})
@Target({ElementType.TYPE_USE, ElementType.TYPE_PARAMETER})
public @interface bottom { }

@SubtypeOf(root.class)
@Target({ElementType.TYPE_USE, ElementType.TYPE_PARAMETER})
public @interface goo { }

@SubtypeOf(goo.class)
@Target({ElementType.TYPE_USE, ElementType.TYPE_PARAMETER})
public @interface foo { }

@SubtypeOf(goo.class)
@Target({ElementType.TYPE_USE, ElementType.TYPE_PARAMETER})
public @interface boo { }

@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.TYPE_USE, ElementType.TYPE_PARAMETER})
@PolymorphicQualifier(goo.class)
public @interface polygoo { }
```

will get you this error.
